tests/unit/io/utils.py

```diff
@@ -103,3 +108,34 @@ def submit_and_wait_for_completion(unit_test, connection, start, end, increment,
     # ensure they are all called back in a timely fashion
     for callback in completed_callbacks:
         unit_test.assertAlmostEqual(callback.expected_wait, callback.get_wait_time(), delta=.15)
+
+
+class TimerConnectionTests(object):
+    def test_multi_timer_validation(self):
+        """
+        Verify that timer timeouts are honored appropriately
+        """
+        # Tests timers submitted in order at various timeouts
+        submit_and_wait_for_completion(self, self.connection_class, 0, 100, 1, 100)
+        # Tests timers submitted in reverse order at various timeouts
+        submit_and_wait_for_completion(self, self.connection_class, 100, 0, -1, 100)
+        # Tests timers submitted in varying order at various timeouts
+        submit_and_wait_for_completion(self, self.connection_class, 0, 100, 1, 100, True),
+
+    def test_timer_cancellation(self):
+        """
+        Verify that timer cancellation is honored
+        """
+
+        # Various lists for tracking callback stage
+        timeout = .1
+        callback = TimerCallback(timeout)
+        timer = self.connection_class.create_timer(timeout, callback.invoke)
+        timer.cancel()
+        # Release context allow for timer thread to run.
+        time.sleep(.2)
+        timer_manager = self.connection_class._timers
+        # Assert that the cancellation was honored
+        self.assertFalse(timer_manager._queue)
+        self.assertFalse(timer_manager._new_timers)
+        self.assertFalse(callback.was_invoked())
```

Comment:
If `utils` gets big in the future it'd be nice to move it to a `mixins` module. No action needed right now, but just something to keep an eye on.